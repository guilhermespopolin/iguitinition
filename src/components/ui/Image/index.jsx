import styled from 'styled-components'
import PropTypes from 'prop-types'

const sizeModifier = ({ size, theme }) => {
  if (size === 'responsive') {
    return `
      width: 100%;
      height: auto;
    `
  }

  return `
    height: ${theme.imageSize[size]};
    width: ${theme.imageSize[size]};
  `
}

const Image = styled.img`
  ${sizeModifier}
`

Image.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large', 'responsive']),
}

Image.defaultProps = {
  size: 'medium',
}

export default Image
