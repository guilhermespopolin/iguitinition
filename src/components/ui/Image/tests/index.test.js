import React from 'react'
import { cleanup } from 'react-testing-library'
import renderComponent from '__tests__/utils'

import Image from '..'

afterEach(cleanup)

describe('<Image />', () => {
  describe('When size modifier equals `small`', () => {
    it('should render properly', () => {
      const { getByAltText } = renderComponent(<Image size="small" alt="image sample" />)
      const imageNode = getByAltText('image sample')

      expect(imageNode).toMatchSnapshot()
    })
  })

  describe('When size modifier equals `medium`', () => {
    it('should render properly', () => {
      const { getByAltText } = renderComponent(<Image size="medium" alt="image sample" />)
      const imageNode = getByAltText('image sample')

      expect(imageNode).toMatchSnapshot()
    })
  })

  describe('When size modifier equals `large`', () => {
    it('should render properly', () => {
      const { getByAltText } = renderComponent(<Image size="large" alt="image sample" />)
      const imageNode = getByAltText('image sample')

      expect(imageNode).toMatchSnapshot()
    })
  })

  describe('When size modifier equals `responsive`', () => {
    it('should render properly', () => {
      const { getByAltText } = renderComponent(<Image size="responsive" alt="image sample" />)
      const imageNode = getByAltText('image sample')

      expect(imageNode).toMatchSnapshot()
    })
  })
})
