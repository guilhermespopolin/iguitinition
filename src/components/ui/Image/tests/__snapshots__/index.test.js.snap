// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> When size modifier equals \`large\` should render properly 1`] = `
.c0 {
  height: 256px;
  width: 256px;
}

<img
  alt="image sample"
  class="c0"
/>
`;

exports[`<Image /> When size modifier equals \`medium\` should render properly 1`] = `
.c0 {
  height: 64px;
  width: 64px;
}

<img
  alt="image sample"
  class="c0"
/>
`;

exports[`<Image /> When size modifier equals \`responsive\` should render properly 1`] = `
.c0 {
  width: 100%;
  height: auto;
}

<img
  alt="image sample"
  class="c0"
/>
`;

exports[`<Image /> When size modifier equals \`small\` should render properly 1`] = `
.c0 {
  height: 32px;
  width: 32px;
}

<img
  alt="image sample"
  class="c0"
/>
`;
